queries.sql

```diff
@@ -1,3 +1,227 @@
+SELECT animals.name,owners.full_name,escape_attempts FROM 
+animals INNER JOIN owners ON animals.owner_id= owners.id INNER JOIN
+ species ON animals.species_id=species.id WHERE
+  escape_attempts=0 AND owners.full_name='Dean Winchester';
+
+  --Who owns the most animals?
+SELECT owners.full_name, COUNT(animals.id) as animal_count
+FROM owners
+INNER JOIN animals ON owners.id = animals.owner_id
+GROUP BY owners.id
+ORDER BY animal_count DESC
+LIMIT 1;
+
+--Who was the last animal seen by William Tatcher?
+SELECT animals.name,visit_date 
+FROM visits 
+JOIN animals ON visits.animal_id = animals.id 
+JOIN vets ON visits.vet_id = vets.id 
+WHERE vets.name = 'William Tatcher' 
+ORDER BY visit_date DESC 
+LIMIT 1;
+
+--How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT animals.name) 
+FROM visits 
+JOIN vets ON visits.vet_id = vets.id 
+JOIN animals ON visits.animal_id = animals.id 
+WHERE vets.name = 'Stephanie Mendez';
+
+--List all vets and their specialIties, including vets with no specialties.
+SELECT vets.name,species.name from specialization
+ RIGHT JOIN vets ON specialization.vet_id = vets.id
+ INNER JOIN species ON specialization.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+
+SELECT animals.name FROM visits 
+JOIN animals ON visits.animal_id=animals.id
+JOIN vets ON visits.vet_id=vets.id WHERE
+ visit_date BETWEEN '2020-04-01' AND
+  '2020-08-30' AND vets.name='Stephanie Mendez';
+--What animal has the most visits to vets?
+SELECT animals.name, COUNT(*) as num_visits
+FROM visits
+JOIN animals ON visits.animal_id = animals.id
+GROUP BY animals.id
+ORDER BY num_visits DESC
+LIMIT 1;
+
+--Who was Maisy Smith's first visit?
+SELECT animals.name, visits.visit_date from visits
+JOIN animals ON visits.animal_id = animals.id
+ORDER BY visit_date ASC
+LIMIT 1;
+
+--Details for most recent visit: animal information, vet information, and date of visit.
+SELECT animals.name AS animal_name, vets.name AS vet_name, visits.visit_date
+FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.id
+ORDER BY visits.visit_date DESC
+LIMIT 1;
+
+--How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) 
+FROM visits 
+JOIN vets ON visits.vet_id = vets.id 
+JOIN animals ON visits.animal_id = animals.id 
+LEFT JOIN specialization ON vets.id = specialization.vet_id 
+AND animals.species_id = specialization.species_id 
+WHERE specialization.vet_id IS NULL;
+
+--What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT s.name AS specialty, COUNT(*) AS visits_count
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN species s ON a.species_id = s.id
+JOIN specialization sp ON sp.species_id = s.id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE a.name = 'Maisy Smith'
+GROUP BY s.name
+ORDER BY visits_count DESC
+LIMIT 1;
```

Comment:
- [ ] This pat of the query is not giving out the right result, let us kindly check and fix it
![Screenshot from 2023-05-04 15-43-04](https://user-images.githubusercontent.com/46388113/236257543-e56bf1d7-e8ce-45f8-9b1a-e48eef85b5f9.png)

Reply:
Hello Sir, may you please give me hint on that last change you requested